tests/unit/test_score_matching.py

```diff
@@ -1088,3 +1055,44 @@ def test_sliced_score_matching_unexpected_num_noise_models(self):
             "int32 and float32",
         )
         # cSpell:enable
+
+
+class TestConvertSteinKernel(unittest.TestCase):
+    """Tests related to the function convert_stein_kernel in score_matching.py."""
+
+    def test_convert_stein_kernel(self) -> None:
+        """Check handling of Stein kernels and standard kernels is consistent."""
+        random_key = jr.key(2_024)
+        dataset_shape = (100, 2)
+
+        dataset = jr.uniform(random_key, dataset_shape)
+        for score_matching in [None, MagicMock()]:
+            for kernel in [PCIMQKernel(), SteinKernel(LinearKernel(), MagicMock())]:
```

Comment:
Should use pytest parameterised tests here, so that the tests can have success/failure reported independently.

Reply:
Agreed, will make the change